code/modules/atmospherics/machinery/unary/pressure_tank.dm

```diff
@@ -0,0 +1,157 @@
+
+/obj/machinery/atmospherics/pipe/tank/oxygen
+	icon = 'icons/obj/atmospherics/tanks/blue_pipe_tank.dmi'
+	name = "Pressure Tank (Oxygen)"
+
+/obj/machinery/atmospherics/pipe/tank/oxygen/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
+/obj/machinery/atmospherics/pipe/tank/nitrogen
+	icon = 'icons/obj/atmospherics/tanks/red_pipe_tank.dmi'
+	name = "Pressure Tank (Nitrogen)"
+
+/obj/machinery/atmospherics/pipe/tank/nitrogen/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.nitrogen = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
+/obj/machinery/atmospherics/pipe/tank/sleeping_agent
+	icon = 'icons/obj/atmospherics/tanks/red_white_pipe_tank.dmi'
+	name = "Pressure Tank (N2O)"
+
+/obj/machinery/atmospherics/pipe/tank/sleeping_agent/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.nitrous_oxide = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
+/obj/machinery/atmospherics/pipe/tank/air
+	icon = 'icons/obj/atmospherics/tanks/white_pipe_tank.dmi'
+	name = "Pressure Tank (Air)"
+
+/obj/machinery/atmospherics/pipe/tank/air/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (50*ONE_ATMOSPHERE*O2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+	air_temporary.nitrogen = (50*ONE_ATMOSPHERE*N2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
+// Experiment for improving the usefulness of air hookups. They have twice the capacity of portable
+// canisters and contain 4 times the volume of their default air mixture (Convair880).
+/obj/machinery/atmospherics/pipe/tank/air_repressurization
+	icon = 'icons/obj/atmospherics/tanks/whitered_pipe_tank.dmi'
+	name = "High-Pressure Tank (Air)"
+	desc = "Large vessel containing a pressurized air mixture for emergency purposes."
+	volume = 2000
+
+/obj/machinery/atmospherics/pipe/tank/air_repressurization/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (180*ONE_ATMOSPHERE*O2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+	air_temporary.nitrogen = (180*ONE_ATMOSPHERE*N2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
+/obj/machinery/atmospherics/pipe/tank/disposing()
+	node1?.disconnect(src)
+	parent = null
+	..()
+
+/obj/machinery/atmospherics/pipe/tank/pipeline_expansion()
+	return list(node1)
```

Comment:
all of this should probably go above the subtypes for this case

Reply:
ight